### PR TITLE
Es6 lib capture perf timings

### PR DESCRIPTION
**Pull request by Calum-Campbell**
## What does this change?
Capture-perf-timing -> es6. 
N.B... Exporting a single function from module so giving it a name and calling it explicitly when it is called in boot.js, similar to this pr https://github.com/guardian/frontend/pull/16473 . 
## What is the value of this and can you measure success?
ES6
## Does this affect other platforms - Amp, Apps, etc?
Shouldn't do

## Tested in CODE?
no

### CODE REVIEW

File: static/src/javascripts/lib/capture-perf-timings.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            timing: parseInt(getMarkTime(mark) || 0, 10),
+        })),
+    };
+
+    ophan.record({
+        performance,
+    });
+};
+
+export { capturePerfTiming };
```
Comment (SiAdcock) [inline static/src/javascripts/lib/capture-perf-timings.js]: Nit: I think this name should be pluralised to match the filename

File: static/src/javascripts/boot.js [hunk 2]
```diff
@@ -75,11 +75,11 @@ const go = () => {
                                                     document.readyState ===
                                                     'complete'
                                                 ) {
-                                                    capturePerfTimings();
+                                                    capturePerfTimings.capturePerfTiming();
```
Comment (SiAdcock) [inline static/src/javascripts/boot.js]: Can we destructure the object on import? Helps with tree shaking, also looks nice in the code:
```
const { capturePerfTimings } = require('lib/capture-perf-timings');

// later
capturePerfTimings();
```

Comment (Calum-Campbell) [inline static/src/javascripts/boot.js]: Yup agree. Looks good, have updated.

File: static/src/javascripts/lib/capture-perf-timings.js (hunk 1 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/capture-perf-timings.js]: Yep good shout.

**Review by SiAdcock — APPROVED** [review]
Love it!

File: static/src/javascripts/lib/capture-perf-timings.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// @flow
+import ophan from 'ophan/ng';
+import { getMarkTime } from 'lib/user-timing';
+import performanceAPI from 'lib/window-performance';
+
+const capturePerfTimings = (): void => {
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/capture-perf-timings.js]: `function capturePerfTimings(): void` reads better.

File: static/src/javascripts/lib/capture-perf-timings.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    const supportsPerformanceProperties =
+        'navigation' in performanceAPI && 'timing' in performanceAPI;
+
+    if (!supportsPerformanceProperties) {
+        return;
+    }
+
+    const timing = performanceAPI && performanceAPI.timing;
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/capture-perf-timings.js]: you can drop the `performanceAPI &&`, implied by the test above.

Comment (Calum-Campbell) [inline static/src/javascripts/lib/capture-perf-timings.js]: Ah yes this is true cheers, will add to future PR.

File: static/src/javascripts/lib/capture-perf-timings.js (hunk 3 above)
Comment (Calum-Campbell) [inline static/src/javascripts/lib/capture-perf-timings.js]: I agree in principle but I think we are going for full conversion to arrow functions where possible.

**Pull request merged**